node/node_handler.go

```diff
@@ -277,6 +304,27 @@ func (node *Node) BroadcastCrosslinkHeartbeatSignalFromBeaconToShards() { // lea
 	}
 }
 
+func createVoters(comm *shard.Committee) map[bls.SerializedPublicKey]struct{} {
+	d := make(map[bls.SerializedPublicKey]struct{})
+	for _, c := range comm.Slots {
+		d[c.BLSPublicKey] = struct{}{}
+	}
+	return d
+}
+
+func getPrivToSign(privs multibls.PrivateKeys, comm *shard.Committee) *bls.PrivateKeyWrapper {
+	var privToSing *bls.PrivateKeyWrapper
+	q := createVoters(comm)
+	for _, priv := range privs {
+		_, found := q[priv.Pub.Bytes]
+		if found { // if key is in committee
```

Comment:
depending on the use case of getPrivToSign, there might be instances where we need the leader key instead of any keys currently elected in the committee